infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/KubernetesInternalRuntime.java

```diff
@@ -574,18 +539,13 @@ protected void internalStop(Map<String, String> stopOptions) throws Infrastructu
     runtimeHangingDetector.stopTracking(getContext().getIdentity());
     if (startSynchronizer.interrupt()) {
       // runtime is STARTING. Need to wait until start will be interrupted properly
-      try {
-        if (!startSynchronizer.awaitInterruption(workspaceStartTimeoutMin, TimeUnit.MINUTES)) {
-          // Runtime is not interrupted yet. It may occur when start was performing by another
-          // Che Server that is crashed so start is hung up in STOPPING phase.
-          // Need to clean up runtime resources
-          probeScheduler.cancel(identity.getWorkspaceId());
-          runtimeCleaner.cleanUp(namespace, identity.getWorkspaceId());
-        }
-      } catch (InterruptedException e) {
-        throw new InfrastructureException(
-            "Interrupted while waiting for start task cancellation", e);
-      }
+
+      // Runtime is not interrupted yet. It may occur when start was performing by another
+      // Che Server that is crashed so start is hung up in STOPPING phase.
+      // Need to clean up runtime resources
+      probeScheduler.cancel(identity.getWorkspaceId());
+      runtimeCleaner.cleanUp(namespace, identity.getWorkspaceId());
+
```

Comment:
could you clarify why do we need the probe sheduler / runtime cleaner cancelation?

Reply:
There is most likely no need for it. I removed the if statement that was surrounding that block of code as it depended on `che.infra.kubernetes.workspace_start_timeout_min` and left the rest with hopes that it would prevent breaking tests. 

I can go back and remove the `setContext` function, or make it perform a no-op.